wfe/web-front-end.go

```diff
@@ -624,13 +625,32 @@ func (wfe *WebFrontEndImpl) RevokeCertificate(response http.ResponseWriter, requ
 	}
 }
 
+func (wfe *WebFrontEndImpl) logCsr(remoteAddr string, cr core.CertificateRequest, registration core.Registration) {
+	var csrLog = struct {
+		RemoteAddr   string
+		CsrBase64    []byte
+		Registration core.Registration
+	}{
+		RemoteAddr:   remoteAddr,
+		CsrBase64:    cr.Bytes,
+		Registration: registration,
+	}
+	_, err := json.Marshal(csrLog)
```

Comment:
To avoid marshaling this twice you could just do

```
err := wfe.log.AuditObject("Certificate request", csrLog)
if err != nil {
		wfe.log.AuditErr(fmt.Errorf(
			"Certificate request failed to serialize. RemoteAddr=%s CSR=%s",
			remoteAddr, base64.StdEncoding.EncodeToString(cr.Bytes)))
}
```